src/Rules.hs

```diff
@@ -90,3 +95,29 @@ buildRules = do
     Rules.Libffi.libffiRules
     packageRules
     Rules.Perl.perlScriptRules
+
+programsStage1Only :: [Package]
+programsStage1Only =
+  [ deriveConstants, genprimopcode, hp2ps, runGhc
```

Comment:
My guess is that most of these will be transitively required anyway. Couldn't we simply ask to build `ghc`?